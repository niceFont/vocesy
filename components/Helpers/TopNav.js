import { Container, Nav, NavDropdown, Navbar, NavbarBrand } from "react-bootstrap"
import Link from "next/link"
import { ExtractName } from "../../lib/utils"
import Cookies from "js-cookie"
import React from "react"
import RoundButton from "./RoundButton"

const TopNav = props => {
	const _logout = () => {
		Cookies.remove("user")
		if (typeof localStorage !== "undefined") {
			localStorage.removeItem("loggedIn")
			window.location.reload()
		}
	}

	return (
		<div style={{
		}} className="justify-content-center">
			<Navbar
				className="topnav"
				style={{
					position: "absolute",
					padding: "0px 10%",
					zIndex: 1,
					width: "100%",
					fontWeight: 500,
					backgroundColor: "none"
				}}
				expand="md"
				variant="light">
				<NavbarBrand>
					<Link href="/">
						<img
							alt="logo"
							style={{
								width: 140,
								height: 55,
							}}
							src="/static/logo_transparent_background.png"
						/>
					</Link>
				</NavbarBrand>
				<Navbar.Toggle className="text-right" style={{
				}} aria-controls="main-menu"></Navbar.Toggle>
				<Navbar.Collapse
					id="main-menu"
					style={{

						padding: 20
					}}>
					<Nav style={{
					}} className="ml-auto justify-content-end">
						<Nav>
							<Nav.Item>
								<Link href="/decks/create">
									<Nav.Link as="a">Create</Nav.Link>
								</Link>
							</Nav.Item>
							<Nav.Item>
								<Link href="/decks">
									<Nav.Link as="a">My Decks</Nav.Link>
								</Link>
							</Nav.Item>
							{!props.user && (
								<Nav>
									<Nav.Item>
										<Link href="/login" replace>
											<Nav.Link as="a">Log in</Nav.Link>
										</Link>
									</Nav.Item>
									<Nav.Item>
										<Link href="/signup">
											<Nav.Link as="a">Sign up</Nav.Link>
										</Link>
									</Nav.Item>
								</Nav>
							)}
							{props.user && (
								<NavDropdown
									title={
										props.user.displayName[0].toUpperCase() +
										props.user.displayName.slice(1)
									}>
									<NavDropdown.Item as="div">
										<Link
											href={`/user/${encodeURI(ExtractName(props.user.displayName)
												.split(" ")
												.join("")
												.toLowerCase())}`}>
											<Nav.Link as="a">Profile</Nav.Link>
										</Link>
									</NavDropdown.Item>
									<NavDropdown.Item
										as="div"
										onClick={_logout}>
										<Nav.Link>Logout</Nav.Link>
									</NavDropdown.Item>
								</NavDropdown>
							)}
						</Nav>
					</Nav>
				</Navbar.Collapse>
			</Navbar>
			<style jsx global>{`
				a {
					cursor: pointer;
				}

			`}</style>
		</div>
	)
}

export default TopNav
