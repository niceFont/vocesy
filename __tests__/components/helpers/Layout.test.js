/*eslint no-undef: off*/
/*eslint camelcase: off*/
import renderer from "react-test-renderer"
import React from "react"
import TopNav from "../../../components/Helpers/TopNav"
//import { mount } from "enzyme"

describe("TopNav", () => {
	it("renders", () => {
		const tree = renderer.create(<TopNav></TopNav>)

		expect(tree.toJSON()).toMatchSnapshot()
	})
})

