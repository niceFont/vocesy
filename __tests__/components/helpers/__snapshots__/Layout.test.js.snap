// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders 1`] = `
Array [
  <nav
    className="navbar navbar-expand-md navbar-light fixed-top"
    style={
      Object {
        "backgroundColor": "white",
        "borderBottom": "1px solid lightgray",
        "paddingLeft": "4%",
        "paddingRight": "4%",
      }
    }
  >
    <span
      className="navbar-brand"
    >
      <img
        alt="logo"
        onClick={[Function]}
        onMouseEnter={[Function]}
        src="/static/logo.png"
        style={
          Object {
            "height": 40,
            "width": 120,
          }
        }
      />
    </span>
    <button
      aria-controls="main-menu"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="main-menu"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <div
          className="nav-item"
        >
          <a
            className="nav-link"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Create
          </a>
        </div>
        <div
          className="nav-item"
        >
          <a
            className="nav-link"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            My Decks
          </a>
        </div>
      </div>
      <div
        className="justify-content-end navbar-nav"
        onKeyDown={[Function]}
      >
        <div
          className="navbar-nav"
          onKeyDown={[Function]}
        >
          <div
            className="navbar-nav"
            onKeyDown={[Function]}
          >
            <div
              className="nav-item"
            >
              <a
                className="nav-link"
                data-rb-event-key={null}
                disabled={false}
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                log in
              </a>
            </div>
            <div
              className="nav-item"
            >
              <a
                className="nav-link"
                data-rb-event-key={null}
                disabled={false}
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                sign up
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>,
  <nav
    className="navbar navbar-expand navbar-dark bg-light fixed-bottom"
  >
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <div
        className="nav-item"
      >
        <span>
          ver. 1.01a Still in Development...
        </span>
      </div>
    </div>
  </nav>,
  <style
    global={true}
    jsx={true}
  >
    
        a {
          cursor: pointer;
        }
      
  </style>,
]
`;
